pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go

```diff
@@ -167,40 +123,27 @@ func (k *klusterletStatusController) sync(ctx context.Context, controllerContext
 		return err
 	}
 
-	// If cluster name is empty, read cluster name from hub config secret
-	clusterName := klusterlet.Spec.ClusterName
-	if clusterName == "" {
-		if hubConfigSecret.Data["cluster-name"] == nil {
-			registrationDegradedCondition.Message = fmt.Sprintf(
-				"Failed to get cluster name from `kubectl get secret -n %q %q -ojsonpath='{.data.cluster-name}`.  This is set by the klusterlet registration deployment.", hubConfigSecret.Namespace, hubConfigSecret.Name)
-			registrationDegradedCondition.Status = metav1.ConditionTrue
-			registrationDegradedCondition.Reason = "ClusterNameMissing"
-			// Work condition will be the same as registration
-			workDegradedCondition.Message = registrationDegradedCondition.Message
-			workDegradedCondition.Status = registrationDegradedCondition.Status
-			workDegradedCondition.Reason = registrationDegradedCondition.Reason
-
+	// If hub kubeconfig does not exist, the bootstrap may not finish, check the bootstrap secret firstly
+	if hubConfigSecret.Data["kubeconfig"] == nil {
+		if registrationDegraded := k.checkBootstrapSecret(ctx, klusterletNS); registrationDegraded != nil {
```

Comment:
I think the bootstrap credentials should always be checked.  I agree that check should not prevent other checks.